components/Weather.js

```diff
@@ -33,72 +54,75 @@ export default class Weather extends React.Component {
+        { enableHighAccuracy: true, timeout: 15000 },
       );
     }
   };
-  _parseWeatherData = (data) => {
+
+  parseWeatherData = (data) => {
     const weather = {};
-    weather.city = datas.city.name;
+    weather.city = data.city.name;
     weather.dataPerHour = [];
-    for (const data of datas.list) {
+    data.forEach((weatherLog) => {
       const perHour = {
-        temp: parseInt(data.main.temp),
-        humidity: data.main.humidity,
-        description: data.weather[0].description,
-        wind: data.wind.speed,
-        hour: data.dt_txt.substr(-8, 5),
-        date: data.dt,
-        icon: data.weather[0].icon,
+        temp: parseInt(weatherLog.main.temp, 10),
+        humidity: weatherLog.main.humidity,
+        description: weatherLog.weather[0].description,
+        wind: weatherLog.wind.speed,
+        hour: weatherLog.dttxt.substr(-8, 5),
+        date: weatherLog.dt,
+        icon: weatherLog.weather[0].icon,
       };
       weather.dataPerHour.push(perHour);
-    }
+    });
     this.setState({
-      weather: weather,
+      weather,
       weatherFetched: true,
     });
   };
-  _fetchDataFromApi = (url) => {
-    const prevThis = this;
 
+  fetchDataFromApi = (url) => {
+    const that = this;
     fetch(url)
       .then((response) => {
         if (response.ok) {
           return response.json();
         }
-        throw new Error('Api returns code ' + response.status);
+        throw new Error(`Api returns code ${response.status}`);
       })
       .then((data) => {
-        prevThis._parseWeatherData(data);
+        that.parseWeatherData(data);
       })
       .catch((error) => {
         console.log(error);
       });
   };
+
   handleCityInput = (e) => {
-    this.setState({cityName: e.nativeEvent.text});
+    this.setState({ cityName: e.nativeEvent.text });
   };
+
   getWeatherByCoords = () => {
     const lat = this.state.latitude;
     const lon = this.state.longitude;
-    const phoneLocated = this.state.phoneLocated;
+    const { phoneLocated } = this.state;
     if (phoneLocated) {
-      let query = `${this._apiAddress}lat=${lat}&lon=${lon}&lang=PL&units=metric&appid=${this._apiKey}`;
-      this._fetchDataFromApi(query);
+      const query = `${this.apiAddress}lat=${lat}&lon=${lon}&lang=PL&units=metric&appid=${this.apiKey}`;
+      this.fetchDataFromApi(query);
     }
   };
+
   getWeatherByCity = () => {
     if (this.state.cityName !== null) {
-      console.log('CityName:');
-      console.log(this.state.cityName);
-      let query = `${this._apiAddress}q=${this.state.cityName}&lang=PL&units=metric&appid=${this._apiKey}`;
-      this._fetchDataFromApi(query);
+      const query = `${this.apiAddress}q=${this.state.cityName}&lang=PL&units=metric&appid=${this.apiKey}`;
+      this.fetchDataFromApi(query);
     }
   };
 
   componentDidMount() {
-    this._getLocation();
+    this.getLocation();
   }
+
   componentDidUpdate(prevProps, prevState) {
```

Comment:
Move it on the top it's connected with component lifecycle. So it's better to have it on the of component declaration.